code/controllers/subsystem/minimaps.dm

```diff
@@ -340,6 +341,9 @@ SUBSYSTEM_DEF(minimaps)
 	RegisterSignal(M, COMSIG_MOVABLE_Z_CHANGED, .proc/on_owner_z_change)
 	if(!SSminimaps.minimaps_by_z["[M.z]"] || !SSminimaps.minimaps_by_z["[M.z]"].hud_image)
 		return
+	if(isAI(M))
+		map = SSminimaps.fetch_minimap_object(default_overwatch_level, minimap_flags)
+		return
```

Comment:
you just need to override the give_action and potentially on_owner_z_change too for /datum/action/minimap/ai. 

You don't need to touch the standard /datum/action/minimap code